crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -476,6 +489,30 @@ impl<'a> SystemParamFetch<'a> for CommandQueue {
     }
 }
 
+/// A system local [`SystemParam`].
+///
+/// A local may only be accessed by the system itself and is therefore not visible to other systems.
+/// If two or more systems specify the same local type each will have their own unique local.
+///
+/// # Examples
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// # let world = &mut World::default();
+/// fn write_to_local(mut local: Local<usize>) {
+///     *local = 42;
+/// }
+/// fn read_from_local(local: Local<usize>) -> usize {
+///     *local
+/// }
+/// let mut write_system = write_to_local.system();
+/// let mut read_system = read_from_local.system();
+/// write_system.initialize(world);
+/// read_system.initialize(world);
+/// write_system.run((), world);
```

Comment:
```suggestion
/// read_system.initialize(world);
/// assert_eq!(read_system.run((), world), 0);
/// write_system.run((), world);
```
This would make it even clearer, that locals aren't shared.